### ISSUE TITLE
Audit all `Display` implementations

### ISSUE THREAD

**Issue by tcharding** [rust-bitcoin/rust-bitcoin#1515]
The `Display` trait includes a functionality for formatting options (padding, alternate format etc.). In order to fully support these options they must be taken into consideration when implementing `Display` - this means the naive method of calling `write!(...)` is almost always wrong since calling `write` uses the `Write` trait (implemented by the formatter) which looses the formatter options.

We should audit every `Display` implementation in the codebase and check if options are correctly implemented. A couple of examples that are broken:
- opcodes (see #1514)
- `OutPoint`

**Comment by Kixunil** [comment]
Right, I think we can ignore errors, at least for now since I don't think we can predict the length for all of them and without it we would have to allocate a temporary `String`.
